tracking/anonymous_tracking.py

```diff
@@ -57,61 +66,61 @@ def init_anonymous_user_id(self):
         return user_id
 
     def send_event(self, name: str, properties: dict = None) -> None:
-        if self.do_not_track:
-            return
+        try:
+            if self.do_not_track:
+                return
 
-        if properties is None:
-            properties = dict()
+            if properties is None:
+                properties = dict()
 
-        properties['run_id'] = self.run_id
-        posthog.capture(distinct_id=self.anonymous_user_id, event=name,
-                        properties={'env': self.env_props, **properties},
-                        groups={'warehouse': self.anonymous_warehouse_id})
+            properties['run_id'] = self.run_id
```

Comment:
I really don't like editing a dict we get as parameter to a method.
It can cause a lot of voodoo and a better practice IMO is to copy it and edit the copy
```python
extended_properties = {**properties, "run_id": self.run_id}
```

Reply:
Yeah, I definitely agree, btw this is not code I touched.